### PR TITLE
[DOC-8549] Document features that require finalization, backward-incompatible changes for v23.2

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-8549] Document features that require finalization, backward-incompatible changes for v23.2

The latter were mostly in place already, but I found a couple missing ones to add.

Source for features that require finalization: https://github.com/cockroachdb/cockroach/blame/release-23.2/pkg/clusterversion/cockroach_versions.go#L448-L506

### HTML previews:
Backward-incompatible changes: [releases/v23.2.md#v23-2-0-backward-incompatible-changes](https://deploy-preview-18278--cockroachdb-docs.netlify.app/docs/releases/v23.2.html#v23-2-0-backward-incompatible-changes)
Features that require finalization: [src/current/v23.2/upgrade-cockroach-version.md#features-that-require-upgrade-finalization](https://deploy-preview-18278--cockroachdb-docs.netlify.app/docs/v23.2/upgrade-cockroach-version.html#features-that-require-upgrade-finalization)

### CODE REVIEW

**Review by fqazi — APPROVED** [review]
The system.regionliveness table changes look good

**Comment by miraradeva** [comment]
The line for #108190 looks good to me.

**Review by maryliag — APPROVED** [review]

**Review by sumeerbhola — APPROVED** [review]
AC changes are fine
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/18278#-:-Nq4LD5l2AaONO0biyfB:b-6i2ct6)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @arulajmani and @miraradeva)

**Comment by mdlinville** [comment]
Thanks for the reviews!

**Review by arulajmani — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/18278)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mdlinville, @mikeCRL, and @miraradeva)

___
*[`src/current/v23.2/upgrade-cockroach-version.md` line 142 at r2](https://reviewable.io/reviews/cockroachdb/docs/18278#-NqUuHJ48tI7VlJbuyrk:-NqUuHJ48tI7VlJbuyrl:b-fuzfij) ([raw file](https://github.com/cockroachdb/docs/blob/b1d3ffe9e5ded07cbd238a33759cc2a11231ea2a/src/current/v23.2/upgrade-cockroach-version.md#L142)):*
> ```Markdown
> When upgrading from {{ previous_version }} to {{ page.version.version }}, certain features and performance improvements will be enabled only after finalizing the upgrade, including but not limited to:
> 
> - The coalescing of storage ranges for each table, index, or partition (collectively referred to as "schema objects") into a single range when individual schema objects are smaller than the default configured maximum range size (controlled using zone configs, specifically the `range_max_bytes parameter`). This change improves scalability with respect to the number of schema objects, since the underlying range count is no longer a potential performance bottleneck. After finalizing the upgrade to v23.2, you may observe a round of range merges and snapshot transfers. To disable this optimization, set the `spanconfig.storage_coalesce_adjacent.enabled` [cluster setting](https://www.cockroachlabs.com/docs/{{ page.version.version }}/cluster-settings) to `false`. Finalization is not required before disabling the optimization. See the [v23.1 release notes]({% link releases/v23.1.md %}) for `SHOW RANGES` for more details. [#102961][#102961]
> ```

nit: instead of saying "finalization is not required", it might be better worded as "this should be done before finalization"

File: src/current/_includes/releases/v23.2/v23.2.0.md [hunk 1]
```diff
@@ -385,11 +385,11 @@ In CockroachDB Self-Hosted, all available features are free to use unless their
...
 
 - The pre-v23.1 output produced by `SHOW RANGES`, `crdb_internal.ranges`, and `crdb_internal.ranges_no_leases` was deprecated in v23.1 and is now replaced by default with output that's compatible with coalesced ranges (anges that pack multiple tables/indexes/partitions into individual ranges). See the [v23.1 release notes]({% link releases/v23.1.md %}) for `SHOW RANGES` for more details. [#102961][#102961]
-- When a deployment is configured to use a time zone (new feature) for log file output using formats `crdb-v1` or `crdb-v2`, it becomes impossible to process the new output log files using the [`cockroach debug merge-logs` command]({% link v23.2/cockroach-debug-merge-logs.md %}) from a previous version. The newest `cockroach debug merge-logs` code must be used instead. [#104265][#104265]
+- When a deployment is configured to use a time zone for log file output using formats `crdb-v1` or `crdb-v2`, new output log entries cannot be processed by notes that have not been upgraded to v23.2. [#104265][#104265]
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.0.md:388]:
```suggestion
- When a deployment is configured to use a time zone for log file output using formats `crdb-v1` or `crdb-v2`, new output log entries cannot be processed by nodes that have not been upgraded to v23.2. [#104265][#104265]
```

File: src/current/_includes/releases/v23.2/v23.2.0.md [hunk 2]
```diff
@@ -385,11 +385,11 @@ In CockroachDB Self-Hosted, all available features are free to use unless their
 Before [upgrading to CockroachDB v23.2]({% link v23.2/upgrade-cockroach-version.md %}), be sure to review the following backward-incompatible changes, as well as key cluster setting changes, and adjust your deployment as necessary.
 
 - The pre-v23.1 output produced by `SHOW RANGES`, `crdb_internal.ranges`, and `crdb_internal.ranges_no_leases` was deprecated in v23.1 and is now replaced by default with output that's compatible with coalesced ranges (anges that pack multiple tables/indexes/partitions into individual ranges). See the [v23.1 release notes]({% link releases/v23.1.md %}) for `SHOW RANGES` for more details. [#102961][#102961]
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.0.md:387]: All of the reference-style links are not resolving as their definitions are not present in this file.

I lean toward leaving the links and copying those definitions into this file, even though we do say to go to the release notes for more information about any bullet. But typically we don't have more detail in the release notes. So I'm OK with keeping these links.

(Side note: perhaps we should go back and add links to docs for key pre-release notes, if we can find an efficient way to do so, right? Something worth considering for 24.1?)

File: src/current/_includes/releases/v23.2/v23.2.0.md [hunk 3]
```diff
@@ -385,11 +385,11 @@ In CockroachDB Self-Hosted, all available features are free to use unless their
 Before [upgrading to CockroachDB v23.2]({% link v23.2/upgrade-cockroach-version.md %}), be sure to review the following backward-incompatible changes, as well as key cluster setting changes, and adjust your deployment as necessary.
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.0.md:385]:
```suggestion
Before [upgrading to CockroachDB v23.2]({% link v23.2/upgrade-cockroach-version.md %}), be sure to review the following backward-incompatible changes, as well as [key cluster setting changes](#v23-2-0-cluster-settings), and adjust your deployment as necessary.
```

**Review by mikeCRL — COMMENTED** [review]
LGTM. A few suggestions.

File: src/current/_includes/releases/v23.2/v23.2.0.md (hunk 3 above)
Comment (mdlinville) [reply src/current/_includes/releases/v23.2/v23.2.0.md]: Done

File: src/current/_includes/releases/v23.2/v23.2.0.md (hunk 1 above)
Comment (mdlinville) [reply src/current/_includes/releases/v23.2/v23.2.0.md]: Done

File: src/current/_includes/releases/v23.2/v23.2.0.md (hunk 2 above)
Comment (mdlinville) [reply src/current/_includes/releases/v23.2/v23.2.0.md:387]: Done. I also added the required links to the upgrade topic. Not sure about adding links to pre-existing notes or what the process should be.

**Review by miraradeva — APPROVED** [review]

**Review by mikeCRL — APPROVED** [review]

**Pull request merged**